examples/encode.py

```diff
@@ -29,17 +31,23 @@ def main():
 
     infile = args.infile
     outfile = args.outfile
-    key = args.key.ljust(16, b'\0')[:16]
+    if args.serial:
+        key = args.serial
+        payload_type = 4
+        digi = True
+    else:
+        key = args.key.ljust(16, b'\0')[:16]
+        payload_type = args.payload_type
+        digi = False
     signature = args.signature
     chunk_size = args.chunk_size
-    payload_type = args.payload_type
     version = args.version << 16
     include_unencrypted_length = args.include_unencrypted_length
 
     data = zcu.compression.compress(infile, chunk_size)
 
-    if payload_type == 2:
-        data = zcu.encryption.aes_encrypt(data, key, chunk_size, include_unencrypted_length)
+    if payload_type == 2 or digi:
```

Comment:
Potentially could use the same `in [2, 4]` logic as in the decode.py?

Reply:
sure